engines/asylum/resources/actor.cpp

```diff
@@ -0,0 +1,4090 @@
+
+						if (getSound()->isPlaying(getWorld()->soundResourceIds[6]))
+							getSound()->stop(getWorld()->soundResourceIds[6]);
+
+						if (getSound()->isPlaying(getWorld()->soundResourceIds[5]))
+							getSound()->stop(getWorld()->soundResourceIds[5]);
+
+						_vm->setGameFlag(kGameFlag1131);
+					} else {
+						getScene()->getActor(11)->updateGraphicData(25);
+						_vm->setGameFlag(kGameFlag556);
+					}
+				}
+			}
+
+			if (_index == getSharedData()->getPlayerIndex()) {
+				if (_frameIndex <= _frameCount - 1) {
+					++_frameIndex;
+				} else {
+					_vm->clearGameFlag(kGameFlag239);
+					getScene()->getActor(10)->updateStatus(kActorStatusEnabled2);
+					hide();
+					_vm->setGameFlag(kGameFlag238);
+
+					// Queue script
+					getScript()->queueScript(getWorld()->getActionAreaById(1000)->scriptIndex, getSharedData()->getPlayerIndex());
+				}
+			}
+
+		} else if (getWorld()->chapter == kChapter11) {
+			if (_index == getSharedData()->getPlayerIndex()) {
+				if (_frameIndex <= _frameCount - 1)
+					++_frameIndex;
+				else
+					resetActors();
+			}
+
+			if (_index >= 10)
+				updateStatus17_Chapter2();
+		}
+		break;
+
+	case kActorStatusAttacking:
+		if (getWorld()->chapter == kChapter2) {
+			if (_index > 12)
+				updateStatus15_Chapter2();
+
+			if (_index == getSharedData()->getPlayerIndex())
+				updateStatus15_Chapter2_Player();
+
+			if (_index == 11)
+				updateStatus15_Chapter2_Actor11();
+
+		} else if (getWorld()->chapter == kChapter11) {
+			if (_index >= 10 && _index < 16)
+				updateStatus15_Chapter11();
+
+			if (_index == getSharedData()->getPlayerIndex())
+				updateStatus15_Chapter11_Player();
+		}
+		break;
+
+	case kActorStatus18:
+		if (getWorld()->chapter == kChapter2) {
+			if (_index > 12)
+				updateStatus18_Chapter2();
+
+			if (_index == 11)
+				updateStatus18_Chapter2_Actor11();
+		}
+		break;
+
+	case kActorStatusDisabled:
+		_frameIndex = (_frameIndex + 1) % _frameCount;
+
+		if (_vm->screenUpdateCount - _lastScreenUpdate > 300) {
+			if (_vm->getRandom(100) < 50) {
+				if (!getSpeech()->getSoundResourceId() || !getSound()->isPlaying(getSpeech()->getSoundResourceId())) {
+					if (isDefaultDirection(10))
+						updateStatus(kActorStatusFidget);
+				}
+			}
+			_lastScreenUpdate = _vm->screenUpdateCount;
+		}
+		break;
+
+	case kActorStatusWalking2:
+		if (getWorld()->chapter == kChapter2) {
+			if (_index > 12) {
+				updateStatus12_Chapter2();
+				return;
+			}
+
+			if (_index == 11) {
+				updateStatus12_Chapter2_Actor11();
+				return;
+			}
+		} else if (getWorld()->chapter == kChapter11) {
+			switch (_index) {
+			default:
+				break;
+
+			case 1:
+				getSpecial()->run(NULL, _index);
+				return;
+
+			case 10:
+			case 11:
+			case 12:
+			case 13:
+			case 14:
+			case 15:
+				updateStatus12_Chapter11();
+				return;
+
+			case 2:
+			case 3:
+			case 4:
+			case 5:
+			case 6:
+			case 7:
+			case 8:
+				return;
+			}
+		}
+		// Fallback to next case
```

Comment:
Compiler Warning:
````
engines/asylum/resources/actor.cpp: In member function ‘void Asylum::Actor::update()’:
engines/asylum/resources/actor.cpp:506:3: warning: this statement may fall through [-Wimplicit-fallthrough=]
   if (getWorld()->chapter == kChapter2) {
   ^~
engines/asylum/resources/actor.cpp:546:2: note: here
  case kActorStatusWalking: {
  ^~~~
````
Again, this looks intentional and thus add // fallthrough comment to surpress the warning.